### PR TITLE
Extension schemas

### PR DESCRIPTION
**Pull request by emackey**
This PR introduces a new means to wire up known glTF extension schemas to VSCode's JSON validation service.  Initially this is just hookup for `KHR` extensions, but in the future this is expected to expand to also cover some well-documented vendor extensions.  (Once this is merged, please open a new issue to request any vendor extension you would like to see covered here).

The new extension schemas are all imported via the `util/importSchema.js` script, the same way the core glTF schemas are, that automates a few key modifications needed for best handling of the schemas in VSCode's JSON validation.  A new lookup table, `util/extensionMap2.0.json` offers the wire-up instructions for connecting particular core glTF schema extension objects to particular extensions that target those objects.  A new script `util/importAll.sh` will re-run the import for all supported schemas at once.

Unknown extensions are still permitted by the editor (although the glTF Validator will still complain about unknown extension prefixes and the like, as expected).  Known extensions will now have complete JSON schema validation, with hover tooltips, auto-complete (CTRL-space), and document problems called out for missing properties and unknown values.

Thanks to @lexaknyazev for a key piece of schema advice that allowed this effort to move forward.

TODO:
- [x] CHANGELOG.md
- [x] README.md

### CODE REVIEW

File: util/importSchema.js [hunk 1]
```diff
@@ -131,12 +143,59 @@ function upgradeDescriptions(data) {
     }
 }
 
-function transformFile(inputFile, outputFile) {
+function transformSharedFolder(data, options) {
+    for (var key in data) {
```
Comment (najadojo) [inline util/importSchema.js]: the 'var's below could be const

**Comment by emackey** [comment]
Work-in-progress: The hand-edits were making me sad too, so I'm migrating them to a legit lookup table.

- [ ] I still need to update the surrounding documentation to reflect this change.

File: schemas/gltf-2.0/extensions/KHR_draco_mesh_compression/node.KHR_draco_mesh_compression.schema.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
```
[the remark this answers is not in the document]
Comment (emackey) [reply schemas/gltf-2.0/extensions/KHR_draco_mesh_compression/node.KHR_draco_mesh_compression.schema.json]: These files are output from the `importSchema.js` script.  Certain values have changed, for example `"$ref" : "glTFid.schema.json"` has been given a working relative path, and other automated changes have been made to improve VSCode's handling of glTF schema files (see `util/README.md` for details).

Might be interesting to submodule the main repo for import purposes, but most developers don't need this.  I expect most people who work on this project just want to build directly from source without re-importing the schemas, and so wouldn't have use for the submodule.

**Comment by emackey** [comment]
OK.  No new hand-edited schemas are being submitted with this PR.  A new look-up table [extensionMap2.0.json](https://github.com/AnalyticalGraphicsInc/gltf-vscode/blob/f24257559362fca239aef1667288f820e7432fee/util/extensionMap2.0.json) has taken the role of helping `importAll.sh` bring in all of the various extension schemas and bind them all together into a coherent whole for VSCode's sake.

**Pull request merged**